src/test/java/org/assertj/core/api/ComparatorFactory_doubleComparatorWithPrecision.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.util.Comparator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class ComparatorFactory_doubleComparatorWithPrecision {
+
+  private final ComparatorFactory INSTANCE = ComparatorFactory.INSTANCE;
+
+  @ParameterizedTest
+  @CsvSource({"0.111, 0.01, 100"})
+  void should_pass_for_expected_equal_to_actual_in_certain_range1(Double expected, Double precision, Double deltaParameter) {
+    Double actual = expected + precision;
+    Comparator<Double> comparator0_1d = INSTANCE.doubleComparatorWithPrecision(precision);
+
+    assertThat(comparator0_1d.compare(expected, actual)).isEqualTo(0);
+    assertThat(comparator0_1d.compare(actual, expected)).isEqualTo(0);
+    assertThat(comparator0_1d.compare(actual + precision / deltaParameter, expected)).isEqualTo(1);
+    assertThat(comparator0_1d.compare(expected, actual + precision / deltaParameter)).isEqualTo(-1);
```

Comment:
the point of parameterized tests is to add parameters value, if you have only one it is basically useless.

the parameters should be: actual, expected and precision:
```java
  @ParameterizedTest
  @CsvSource({"0.111, 0.01, 100",
              "0.01, 0.111, 100",
              // add as many combination you can think of to cover edge cases
              "0.01, 0.111, 100"})
  void should_evaluate_actual_and_expected_as_equals(Double actual, Double expected, Double precision) {
    // GIVEN
    Comparator<Double> comparator = INSTANCE.doubleComparatorWithPrecision(precision);
    // WHEN/THEN
    then(comparator.compare(actual, expected)).isZero();
}
```
let me know if you don't understand.

Reply:
This is a good suggestion! I have understood it, and will work on it.